scripts/globals/events/harvest_festival.lua

```diff
@@ -0,0 +1,318 @@
+        { name = "Trick_Bones",    costume = 0 },
+        { name = "Trick_Skeleton", costume = 0 },
+    },
+
+    [xi.zone.SOUTHERN_SAN_DORIA] =
+    {
+        { name = "Machielle",  costume = 580 }, -- Yagudo
+        { name = "Phamelise",  costume = 368 }, -- Ghost
+        { name = "Apairemant", costume = 531 }, -- Dark Stalker
+        { name = "Corua",      costume = 564 }, -- Skeleton
+        { name = "Pourette",   costume = 612 }, -- Orc
+    },
+
+    [xi.zone.WINDURST_WOODS] =
+    {
+        { name = "Meriri",          costume = 644 }, -- Quadav
+        { name = "Kuzah_Hpirohpon", costume = 580 }, -- Yagudo
+        { name = "Taraihi-Perunhi", costume = 368 }, -- Ghost
+        { name = "Nhobi_Zalkia",    costume = 544 }, -- Shade
+        { name = "Millerovieunet",  costume = 612 }, -- Orc
+    },
+
+    [xi.zone.WINDURST_WATERS] =
+    {
+        { name = "Upih_Khachla",   costume = 368 }, -- Ghost
+        { name = "Ness_Rugetomal", costume = 365 }, -- Hound
+        { name = "Maqu_Molpih",    costume = 612 }, -- Orc
+        { name = "Ahyeekih",       costume = 540 }, -- Shade
+        { name = "Ensasa",         costume = 564 }, -- Skeleton
+        { name = "Trick_Spirit",   costume = 0 },
+        { name = "Trick_Specter",  costume = 0 },
+        { name = "Trick_Shadow",   costume = 0 },
+        { name = "Trick_Shade",    costume = 0 },
+        { name = "Trick_Phantom",  costume = 0 },
+        { name = "Trick_Ghost",    costume = 0 },
+        { name = "Trick_Wight",    costume = 0 },
+        { name = "Trick_Ghast",    costume = 0 },
+        { name = "Trick_Bones",    costume = 0 },
+        { name = "Trick_Skeleton", costume = 0 },
+    },
+
+    [xi.zone.BASTOK_MINES] =
+    {
+        { name = "Faustin",      costume = 365 }, -- Hound
+        { name = "Mille",        costume = 644 }, -- Quadav
+        { name = "Aulavia",      costume = 368 }, -- Ghost
+        { name = "Proud_Beard",  costume = 544 }, -- Shade
+        { name = "Emaliveulaux", costume = 564 }, -- Skeleton
+    },
+
+    [xi.zone.BASTOK_MARKETS] =
+    {
+        { name = "Olwyn",          costume = 368 }, -- Ghost
+        { name = "Trick_Spirit",   costume = 0 },
+        { name = "Trick_Specter",  costume = 0 },
+        { name = "Trick_Shadow",   costume = 0 },
+        { name = "Trick_Shade",    costume = 0 },
+        { name = "Trick_Phantom",  costume = 0 },
+        { name = "Trick_Ghost",    costume = 0 },
+        { name = "Trick_Wight",    costume = 0 },
+        { name = "Trick_Ghast",    costume = 0 },
+        { name = "Trick_Bones",    costume = 0 },
+        { name = "Trick_Skeleton", costume = 0 },
+    },
+}
+
+local costumeList =
+{
+    math.random(531, 534), -- Stalker
+    math.random(580, 607), -- Yagudo
+    math.random(644, 671), -- Quadav
+    math.random(535, 538), -- Shade
+    math.random(707, 711), -- Gigas
+    math.random(740, 756), -- Demon
+    math.random(612, 639), -- Orc
+    math.random(484, 511), -- Goblin
```

Comment:
These will only be randomised once when this table is created. If 3 people roll on this table and each get Yagudo, they'll all get the same model - not random ones. You're better off totally unrolling all the randoms so the whole ranges are exposed.

Reply:
What I did instead is define the range and select a random value in said range whenever the function is called. Saves having dozens of numbers in a long list